Added documentation on docker usage (#1865)
prometheus/alertmanager: README.md

$ make build BINARIES=amtool
```

## Docker

> <!--- example of complete docker run command to run a container --->

Reviewer: How about linking to the `Procfile` of this repository, as it gives a user a full setup on their local machine.

Reviewer: https://github.com/prometheus/alertmanager/blob/master/Procfile

Author: I'm talking about an actual invokation of `docker run ...`, using volumes(`-v`) and port(`-p`) mapping as appropriate, then the command arguments. The `Procfile` contains example invokations of the `alertmanager` binary, but it will look different when using `docker run`, so it can be confusing to people less familiar with docker and unix.

<!--- example of a docker-compose.yml service snippet --->

### Ports

> * `9093` : main api port

Reviewer: ```suggestion
* `9093`: main api port
* `9094`: cluster port 
```

### Volumes

* `/alertmanager`: storage directory

> These paths can also be mapped to volumes:

Reviewer: I am a bit confused by this comment as `/alertmanager` can also be mapped to volumes.

Author: `/alertmanager` is declared as a volume in the Dockerfile, so will always be a volume. But any file or directory can be mounted as a volume. I thought a good idea to mention files and directories that people likely also want to mount as volumes. 

I guess the wording can be changed to something like 
```suggestion
You might also want to use volumes for these paths:
```

* `/etc/alertmanager/alertmanager.yml`: default configuration file path

> ### Environment variables

Reviewer: I am not aware of any. Configuration is limited to the config file and the command line flags.

Reviewer 2: Same for me. I'd remove this section.

$ make build BINARIES=amtool
```

## Docker

<!--- example of complete docker run command to run a container --->
<!--- example of a docker-compose.yml service snippet --->

> ### Ports

Reviewer 2: Exposed ports?

* `9093`: main api port
* `9094`: cluster port 

### Volumes

* `/alertmanager`: storage directory

These paths can also be mapped to volumes:
* `/etc/alertmanager/alertmanager.yml`: default configuration file path

### Environment variables

<!--- Add documentation here --->

> ### Command and entrypoint

Reviewer 2: "Entrypoint" alone?

The container entrypoint is the [`alertmanager` command](https://prometheus.io/docs/alerting/configuration/)<!---More complete documentation of the command would be useful here--->. 

> The default docker command passed to the entrypoint is `--config.file=/etc/alertmanager/alertmanager.yml --storage.path=/alertmanager`. Specifying the docker command(using `docker run quay.io/prometheus/alertmanager $COMMAND`) thus override those default arguments.

Reviewer 2: Regarding this paragraph, I'm not sure we want to get into this level of details as 1) the arguments may change over time and 2) it is more a matter of the container runtime.

Author: Fair enough, we could link to a documentation page for the binary.
